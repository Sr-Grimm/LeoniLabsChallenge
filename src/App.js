import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import SearchPage from './Components/screens/SearchPage';
import DetailPage from './Components/screens/DetailPage';
import Header from './Components/templates/Header';
import Footer from './Components/templates/Footer';
import Container from "./Components/atoms/Container";

export default function App() {
  return (
  <Router>
    <Redirect from="/" to="/searcher" />
    <Container flexD="column" bgColor="black" width="100%" height="100%">
      <Header></Header>
      <Container width="100%" height="85%">
        <Switch>
          <Route  path="/searcher">
            <SearchPage/>
          </Route>
          <Route path="/detail/:id">
            <DetailPage />
          </Route>
        </Switch>
      </Container>
      <Footer></Footer>
    </Container>
  </Router>
  );
}