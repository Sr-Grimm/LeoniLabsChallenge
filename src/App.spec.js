import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders LeoniLabs Searcher header', () => {
  const component = render(<App />);
  expect(component).toMatchSnapshot();
});

