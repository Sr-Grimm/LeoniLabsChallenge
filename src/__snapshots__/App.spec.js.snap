// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LeoniLabs Searcher header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm coTIdm"
        height="100%"
        overflow="hidden"
        width="100%"
      >
        <div
          class="sc-AxjAm iNJtNO"
          height="10%"
          overflow="hidden"
          width="100%"
        >
          <p
            class="sc-AxhCb ksoRvs"
            color="white"
            text-decoration="none"
          >
            LeoniLabs Searcher
          </p>
        </div>
        <div
          class="sc-AxjAm javIoX"
          height="85%"
          overflow="hidden"
          width="100%"
        >
          <div
            class="sc-AxjAm ciusbb"
            height="100%"
            overflow="hidden"
            width="100%"
          >
            <div
              class="sc-AxjAm bJyHPr"
              height="100%"
              overflow="hidden"
              width="20%"
            >
              <div
                class="sc-AxjAm ejTeOA"
                height="auto"
                overflow="hidden"
                width="auto"
              >
                <input
                  class="sc-AxiKw eCHmLl"
                  type="checkbox"
                />
                <p
                  class="sc-AxhCb eCnQFw"
                  color="white"
                  text-decoration="none"
                >
                  Advanced Search
                </p>
              </div>
              <div
                class="sc-AxjAm irLdAX"
                height="auto"
                overflow="hidden"
                width="auto"
              >
                <div
                  class="sc-AxjAm facoKZ"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <input
                    class="sc-AxgMl jTUUmS"
                    placeholder="Search here"
                    type="text"
                    value=""
                    width="90%"
                  />
                </div>
                <div
                  class="sc-AxjAm ccTrau"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <div
                    class="sc-AxjAm jXzRHe"
                    height="auto"
                    overflow="hidden"
                    width="auto"
                  >
                    <input
                      class="sc-AxhUy dNxuhj"
                      name="simpleValue"
                      type="radio"
                      value="first_name"
                    />
                    <p
                      class="sc-AxhCb dSZRKE"
                      color="white"
                      text-decoration="none"
                    >
                      First name
                    </p>
                  </div>
                  <div
                    class="sc-AxjAm jXzRHe"
                    height="auto"
                    overflow="hidden"
                    width="auto"
                  >
                    <input
                      class="sc-AxhUy dNxuhj"
                      name="simpleValue"
                      type="radio"
                      value="last_name"
                    />
                    <p
                      class="sc-AxhCb dSZRKE"
                      color="white"
                      text-decoration="none"
                    >
                      Last name
                    </p>
                  </div>
                  <div
                    class="sc-AxjAm jXzRHe"
                    height="auto"
                    overflow="hidden"
                    width="auto"
                  >
                    <input
                      class="sc-AxhUy dNxuhj"
                      name="simpleValue"
                      type="radio"
                      value="middle_name"
                    />
                    <p
                      class="sc-AxhCb dSZRKE"
                      color="white"
                      text-decoration="none"
                    >
                      Middle name
                    </p>
                  </div>
                  <div
                    class="sc-AxjAm jXzRHe"
                    height="auto"
                    overflow="hidden"
                    width="auto"
                  >
                    <input
                      class="sc-AxhUy dNxuhj"
                      name="simpleValue"
                      type="radio"
                      value="born_date"
                    />
                    <p
                      class="sc-AxhCb dSZRKE"
                      color="white"
                      text-decoration="none"
                    >
                      Birth date
                    </p>
                  </div>
                  <div
                    class="sc-AxjAm jXzRHe"
                    height="auto"
                    overflow="hidden"
                    width="auto"
                  >
                    <input
                      class="sc-AxhUy dNxuhj"
                      name="simpleValue"
                      type="radio"
                      value="party"
                    />
                    <p
                      class="sc-AxhCb dSZRKE"
                      color="white"
                      text-decoration="none"
                    >
                      Party
                    </p>
                  </div>
                </div>
              </div>
              <button
                class="sc-AxirZ cbhsUY"
                color="white"
                height="auto"
                label="Search"
                width="auto"
              >
                Search
              </button>
            </div>
            <div
              class="sc-AxjAm iTRvSA"
              height="100%"
              overflow="hidden"
              width="80%"
            >
              <div
                class="sc-AxheI lbAgMF"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxjAm Lmhlf"
          height="5%"
          overflow="hidden"
          width="100%"
        >
          <p
            class="sc-AxhCb ibhECr"
            color="white"
            text-decoration="none"
          >
            Create react app used
          </p>
          <div
            class="sc-AxjAm gsQusT"
            height="auto"
            overflow="hidden"
            width="auto"
          >
            <p
              class="sc-AxhCb sqiZv"
              color="white"
              text-decoration="none"
            >
              Created by Agustin Fredes
            </p>
            <a
              class="sc-AxmLO iXUPpc"
              href="https://www.linkedin.com/in/agustinfredes/"
              target="_blank"
            >
              <p
                class="sc-AxhCb ejFRSo"
                color="#0e76a8"
                text-decoration="none"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  />
                </svg>
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm coTIdm"
      height="100%"
      overflow="hidden"
      width="100%"
    >
      <div
        class="sc-AxjAm iNJtNO"
        height="10%"
        overflow="hidden"
        width="100%"
      >
        <p
          class="sc-AxhCb ksoRvs"
          color="white"
          text-decoration="none"
        >
          LeoniLabs Searcher
        </p>
      </div>
      <div
        class="sc-AxjAm javIoX"
        height="85%"
        overflow="hidden"
        width="100%"
      >
        <div
          class="sc-AxjAm ciusbb"
          height="100%"
          overflow="hidden"
          width="100%"
        >
          <div
            class="sc-AxjAm bJyHPr"
            height="100%"
            overflow="hidden"
            width="20%"
          >
            <div
              class="sc-AxjAm ejTeOA"
              height="auto"
              overflow="hidden"
              width="auto"
            >
              <input
                class="sc-AxiKw eCHmLl"
                type="checkbox"
              />
              <p
                class="sc-AxhCb eCnQFw"
                color="white"
                text-decoration="none"
              >
                Advanced Search
              </p>
            </div>
            <div
              class="sc-AxjAm irLdAX"
              height="auto"
              overflow="hidden"
              width="auto"
            >
              <div
                class="sc-AxjAm facoKZ"
                height="auto"
                overflow="hidden"
                width="auto"
              >
                <input
                  class="sc-AxgMl jTUUmS"
                  placeholder="Search here"
                  type="text"
                  value=""
                  width="90%"
                />
              </div>
              <div
                class="sc-AxjAm ccTrau"
                height="auto"
                overflow="hidden"
                width="auto"
              >
                <div
                  class="sc-AxjAm jXzRHe"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <input
                    class="sc-AxhUy dNxuhj"
                    name="simpleValue"
                    type="radio"
                    value="first_name"
                  />
                  <p
                    class="sc-AxhCb dSZRKE"
                    color="white"
                    text-decoration="none"
                  >
                    First name
                  </p>
                </div>
                <div
                  class="sc-AxjAm jXzRHe"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <input
                    class="sc-AxhUy dNxuhj"
                    name="simpleValue"
                    type="radio"
                    value="last_name"
                  />
                  <p
                    class="sc-AxhCb dSZRKE"
                    color="white"
                    text-decoration="none"
                  >
                    Last name
                  </p>
                </div>
                <div
                  class="sc-AxjAm jXzRHe"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <input
                    class="sc-AxhUy dNxuhj"
                    name="simpleValue"
                    type="radio"
                    value="middle_name"
                  />
                  <p
                    class="sc-AxhCb dSZRKE"
                    color="white"
                    text-decoration="none"
                  >
                    Middle name
                  </p>
                </div>
                <div
                  class="sc-AxjAm jXzRHe"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <input
                    class="sc-AxhUy dNxuhj"
                    name="simpleValue"
                    type="radio"
                    value="born_date"
                  />
                  <p
                    class="sc-AxhCb dSZRKE"
                    color="white"
                    text-decoration="none"
                  >
                    Birth date
                  </p>
                </div>
                <div
                  class="sc-AxjAm jXzRHe"
                  height="auto"
                  overflow="hidden"
                  width="auto"
                >
                  <input
                    class="sc-AxhUy dNxuhj"
                    name="simpleValue"
                    type="radio"
                    value="party"
                  />
                  <p
                    class="sc-AxhCb dSZRKE"
                    color="white"
                    text-decoration="none"
                  >
                    Party
                  </p>
                </div>
              </div>
            </div>
            <button
              class="sc-AxirZ cbhsUY"
              color="white"
              height="auto"
              label="Search"
              width="auto"
            >
              Search
            </button>
          </div>
          <div
            class="sc-AxjAm iTRvSA"
            height="100%"
            overflow="hidden"
            width="80%"
          >
            <div
              class="sc-AxheI lbAgMF"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-AxjAm Lmhlf"
        height="5%"
        overflow="hidden"
        width="100%"
      >
        <p
          class="sc-AxhCb ibhECr"
          color="white"
          text-decoration="none"
        >
          Create react app used
        </p>
        <div
          class="sc-AxjAm gsQusT"
          height="auto"
          overflow="hidden"
          width="auto"
        >
          <p
            class="sc-AxhCb sqiZv"
            color="white"
            text-decoration="none"
          >
            Created by Agustin Fredes
          </p>
          <a
            class="sc-AxmLO iXUPpc"
            href="https://www.linkedin.com/in/agustinfredes/"
            target="_blank"
          >
            <p
              class="sc-AxhCb ejFRSo"
              color="#0e76a8"
              text-decoration="none"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                />
              </svg>
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
